### PR TITLE
JENKINS-36207# Make BluePipeline extensible

### PR DESCRIPTION
**Pull request by vivek**
Related to issue # . 

@michaelneale @imeredith 

@reviewbybees

### CODE REVIEW

**Comment by blueoceandeploy** [comment]
Can one of the admins verify this patch?

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineImpl.java [hunk 1]
```diff
@@ -164,4 +177,28 @@ protected static String getRecursivePathFromFullName(BluePipeline pipeline){
         return pipelinePath.toString();
     }
 
+    @Extension(ordinal = 1)
+    public static class PipelineFactoryImpl extends BluePipelineFactory {
```
Comment (michaelneale) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineImpl.java]: name could be maybe less redundant? DefaultPipelineFactory? Or just PipelineFactory?

Comment (vivek) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineImpl.java]: Well, this is inner class, implementation name, does it matter to be so specific about naming? I would worry about it if it was an interface or contract like what goes in blueocean-rest module.

Comment (michaelneale) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineImpl.java]: 👍🏻

**Comment by michaelneale** [comment]
🐝

**Pull request merged**